go/service/user.go

```diff
@@ -651,6 +675,15 @@ func (h *UserHandler) CanLogout(ctx context.Context, sessionID int) (res keybase
 	})
 
 	if err != nil {
+		isRevoked, err2 := isActiveDeviceRevoked(libkb.NewMetaContext(ctx, h.G()))
```

Comment:
streamline error handling here and make it more go-y.  don't have nested-if error conditions

Reply:
I wanted it to be nested here so this is only checked if call to hasRandomPW fails, because hasRandomPW can be cached and works offline. So if `CheckCurrentUIDDeviceID` is only called when hasRandomPW fails, the whole RPC still have the ability to work offline (assuming `hasRandomPW` can find a cached value).

Unless there is some other control flow simplification that I'm missing